File: app/src/main/java/ie/dublinbuspal/android/data/local/dao/BusStopDao.java
package ie.dublinbuspal.android.data.local.dao;


import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Transaction;
import ie.dublinbuspal.android.data.local.DbInfo;
import ie.dublinbuspal.android.data.local.entity.BusStop;

@Dao
public abstract class BusStopDao implements BaseDao<BusStop> {

    @Query(DbInfo.SELECT_BUS_STOPS)
    public abstract List<BusStop> selectAll();

    @Query(DbInfo.DELETE_BUS_STOPS)
    public abstract void deleteAll();

    @Transaction
    public void replaceAll(List<BusStop> busStops) {
        deleteAll();
        insertAll(busStops);
    }

}

File: database/src/main/java/ie/dublinbuspal/database/dao/BusStopDao.kt
package ie.dublinbuspal.database.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import ie.dublinbuspal.database.entity.BusStopEntity
import io.reactivex.Maybe

@Dao
interface BusStopDao {

    @Query("SELECT * FROM stops")
    fun selectAll(): Maybe<List<BusStopEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(entities: List<BusStopEntity>)

    @Query("DELETE FROM stops")
    fun deleteAll()

}
